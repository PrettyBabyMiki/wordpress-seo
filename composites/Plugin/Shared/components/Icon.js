import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import _omit from "lodash/omit";

/**
 * Returns the Icon component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} Icon component.
 */
export const Icon = ( props ) => {
	const IconComponent = styled( props.icon )`
		width: ${ props.size };
		height: ${ props.size };
		fill: ${ props.color };
		flex: none;
	`;

	// Remove the props that are no longer needed.
	const newProps = _omit( props, [ "icon", "size", "color" ] );

	return <IconComponent role="img" aria-hidden="true" focusable="false" { ...newProps } />;
};

Icon.propTypes = {
	icon: PropTypes.func.isRequired,
	color: PropTypes.string.isRequired,
	size: PropTypes.string,
};

Icon.defaultProps = {
	size: "16px",
};

/**
 * Returns the IconWithAriaLabel component.
 *
 * Safari 11 doesn't expose to browsers the ARIA role set on SVG elements.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} IconWithAriaLabel component.
 */
export const IconWithAriaLabel = ( props ) => {
	return <span role="img" aria-label={ props.ariaLabel }>
		<Icon
			icon={ props.icon }
			color={ props.color }
			size={ props.size }
			role="presentation"
		/>
	</span>;
};

IconWithAriaLabel.propTypes = {
	icon: PropTypes.func.isRequired,
	color: PropTypes.string.isRequired,
	size: PropTypes.string.isRequired,
	ariaLabel: PropTypes.string.isRequired,
};
