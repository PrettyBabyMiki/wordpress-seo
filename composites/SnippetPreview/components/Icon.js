import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

/**
 * Returns the Icon component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} Icon component.
 */
export const Icon = ( props ) => {
	const IconComponent = styled( props.icon )`
		float: left;
		width: ${ props => props.iconSize }px;
		height: ${ props => props.iconSize }px;
		fill: ${ props => props.iconColor };
	`;

	return <IconComponent aria-hidden="true" iconSize={ props.iconSize } iconColor={ props.iconColor } />;
};

Icon.propTypes = {
	icon: PropTypes.func.isRequired,
	iconColor: PropTypes.string.isRequired,
	iconSize: PropTypes.number,
};

Icon.defaultProps = {
	iconSize: 16,
};
