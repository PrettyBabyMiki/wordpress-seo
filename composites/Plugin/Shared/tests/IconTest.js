import React from "react";
import renderer from "react-test-renderer";

import Icon from "../components/Icon";

// Replace this Fake component with a real component, when there will be one.
const CircleIcon = ( props ) => {
	/* eslint-disable max-len */
	return(
		<svg { ...props } width="1792" height="1792" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
			<path d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z" />
		</svg>
	);
	/* eslint-enable max-len */
};

test( "the Icon matches the snapshot", () => {
	const component = renderer.create(
		<Icon icon={ CircleIcon } color="red" size="44px" />
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
