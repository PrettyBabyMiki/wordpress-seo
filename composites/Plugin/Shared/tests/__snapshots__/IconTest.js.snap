// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Icon matches the snapshot 1`] = `
.c0 {
  width: 44px;
  height: 44px;
  fill: red;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<svg
  aria-hidden="true"
  className="c0"
  focusable="false"
  height="1792"
  role="img"
  viewBox="0 0 1792 1792"
  width="1792"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
  />
</svg>
`;
